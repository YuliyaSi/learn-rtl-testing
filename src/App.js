import {useState} from "react";
import App1 from "./App1";
import App2 from "./App2";

const Search = ({ value, onChange, children }) => (
    <div>
        <img alt="search-image"/>
      <label htmlFor="search">{children}</label>
        <br/>
      <input type="text" id="search" value={value} onChange={onChange} placeholder="Search text..."/>
    </div>
)

const App = () => {
  const [search, setSearch] = useState('');

  const handleChange = ({ target }) => {
    setSearch(target.value);
  }

  return (
      <>
          <div>
              <Search value={search} onChange={handleChange}>
                  Search:
              </Search>
              <p>Searches for {search || '...'}</p>
          </div>
          <hr/>
          <App1/>
          <hr/>
          <App2/>
      </>
  )
}

export default App;