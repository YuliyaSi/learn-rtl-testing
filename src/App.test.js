import { render, screen } from '@testing-library/react';
import App from './App';

// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });
//
// test('render link', () => {
//   const { getByText } = render(<App/>);
//   const linkElement = getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// })

test('show DOM in console with screen', () => {
  render(<App/>);
  screen.debug()
})

test('use fragment', () => {
  const { asFragment } = render(<App/>);
  expect(asFragment(<App/>)).toMatchSnapshot();
})
