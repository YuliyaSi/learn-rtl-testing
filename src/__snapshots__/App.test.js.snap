// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`use fragment 1`] = `
<DocumentFragment>
  <div>
    <div>
      <label
        for="search"
      >
        Search:
      </label>
      <br />
      <input
        id="search"
        type="text"
        value=""
      />
    </div>
    <p>
      Searches for ...
    </p>
  </div>
</DocumentFragment>
`;
